djstripe/views.py

```diff
@@ -131,15 +132,35 @@ def post(self, request, *args, **kwargs):
 #                              Subscription Views                              #
 # ============================================================================ #
 
-
-class SubscribeFormView(LoginRequiredMixin, FormValidMessageMixin, SubscriptionMixin, FormView):
+class ConfirmFormView(LoginRequiredMixin, FormValidMessageMixin, SubscriptionMixin, FormView):
     """TODO: Add stripe_token to the form and use form_valid() instead of post()."""
 
     form_class = PlanForm
-    template_name = "djstripe/subscribe_form.html"
+    template_name = "djstripe/confirm_form.html"
     success_url = reverse_lazy("djstripe:history")
     form_valid_message = "You are now subscribed!"
 
+    def get(self, request, *args, **kwargs):
+        plan_slug = self.kwargs['plan']
+        if plan_slug not in PAYMENT_PLANS:
+            return redirect("djstripe:subscribe")
+
+        plan = PAYMENT_PLANS[plan_slug]
+        customer, created = Customer.get_or_create(
+            subscriber=subscriber_request_callback(self.request))
+
+        if customer.current_subscription.plan == plan['plan'] and customer.current_subscription.status != CurrentSubscription.STATUS_CANCELLED:
```

Comment:
an other issue here with new subscription, need to replace by:

    if hasattr(customer, "current_subscription") and customer.current_subscription.plan == plan['plan'] and customer.current_subscription.status != CurrentSubscription.STATUS_CANCELLED: